diff --git a/Makefile b/Makefile
index c880c47..6359236 100644
--- a/Makefile
+++ b/Makefile
@@ -134,16 +134,22 @@ print-version:
 .c.o:
 	$(CC) $(CFLAGS) $(ALL_CPPFLAGS) -c -o $@ $<
 
+$(LOBJS):
+	$(CC) $(CFLAGS) $(ALL_CPPFLAGS) -fPIC -c -o $@ $<
+
 LIBST_OBJS = sam_opts.o sam_utils.o bedidx.o
 
 
-lib:libbam.a
+lib:libbam.so.1.1
+
+libbam.so.1.1:$(LOBJS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -shared -Wl,--soname,$@ -o $@ $(LOBJS) $(HTSLIB)
+	ln -sf $@ libbam.so.1
+	ln -sf $@ libbam.so
 
-libbam.a:$(LOBJS)
-	$(AR) -csru $@ $(LOBJS)
 
-samtools: $(AOBJS) $(LZ4OBJS) libbam.a libst.a $(HTSLIB)
-	$(CC) $(ALL_LDFLAGS) -o $@ $(AOBJS) $(LZ4OBJS) libbam.a libst.a $(HTSLIB_LIB) $(CURSES_LIB) -lm $(ALL_LIBS) -lpthread
+samtools: $(AOBJS) $(LZ4OBJS) libbam.so.1.1 libst.a $(HTSLIB)
+	$(CC) $(ALL_LDFLAGS) -o $@ $(AOBJS) $(LZ4OBJS) libbam.so.1.1 libst.a $(HTSLIB_LIB) $(CURSES_LIB) -lm $(ALL_LIBS) -lpthread
 
 # For building samtools and its test suite only: NOT to be installed.
 libst.a: $(LIBST_OBJS)
