--- Makefile.orig	2018-11-12 14:30:25.837796124 -0700
+++ Makefile	2018-11-12 14:40:34.173969357 -0700
@@ -132,16 +132,20 @@
 .c.o:
 	$(CC) $(CFLAGS) $(ALL_CPPFLAGS) -c -o $@ $<
 
-LIBST_OBJS = sam_opts.o sam_utils.o
+$(LOBJS):
+	$(CC) $(CFLAGS) $(ALL_CPPFLAGS) -fPIC -c -o $@ $<
 
+LIBST_OBJS = sam_opts.o sam_utils.o
 
-lib:libbam.a
+lib: libbam.so.1.1
 
-libbam.a:$(LOBJS)
-	$(AR) -csru $@ $(LOBJS)
+libbam.so.1.1:$(LOBJS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -shared -Wl,--soname,$@ -o $@ $(LOBJS) $(HTSLIB)
+	ln -sf $@ libbam.so.1
+	ln -sf $@ libbam.so
 
-samtools: $(AOBJS) $(LZ4OBJS) libbam.a libst.a $(HTSLIB)
-	$(CC) $(ALL_LDFLAGS) -o $@ $(AOBJS) $(LZ4OBJS) libbam.a libst.a $(HTSLIB_LIB) $(CURSES_LIB) -lm $(ALL_LIBS) -lpthread
+samtools: $(AOBJS) $(LZ4OBJS) libbam.so.1.1 libst.a $(HTSLIB)
+	$(CC) $(ALL_LDFLAGS) -o $@ $(AOBJS) $(LZ4OBJS) libbam.so.1.1 libst.a $(HTSLIB_LIB) $(CURSES_LIB) -lm $(ALL_LIBS) -lpthread
 
 # For building samtools and its test suite only: NOT to be installed.
 libst.a: $(LIBST_OBJS)
