diff --git a/include/objects/seqloc/Seq_id.hpp b/include/objects/seqloc/Seq_id.hpp
index 4367c7c..a6ca629 100644
--- a/include/objects/seqloc/Seq_id.hpp
+++ b/include/objects/seqloc/Seq_id.hpp
@@ -748,9 +748,9 @@ public:
                        TComposeOSLTFlags parse_flags = 0) const;
 
     /// ID length restrictions
-    const static size_t kMaxLocalIDLength    = 50;
+    const static size_t kMaxLocalIDLength    = 120;
     const static size_t kMaxGeneralDBLength  = 20;
-    const static size_t kMaxGeneralTagLength = 50;
+    const static size_t kMaxGeneralTagLength = 120;
     const static size_t kMaxAccessionLength  = 30;
 
 private:
diff --git a/src/build-system/ncbi_package_version b/src/build-system/ncbi_package_version
index 46b81d8..9ff167d 100644
--- a/src/build-system/ncbi_package_version
+++ b/src/build-system/ncbi_package_version
@@ -1 +1 @@
-2.11.0
+2.11.99
diff --git a/src/objtools/readers/seqid_validate.cpp b/src/objtools/readers/seqid_validate.cpp
index dc33e4f..8daf286 100644
--- a/src/objtools/readers/seqid_validate.cpp
+++ b/src/objtools/readers/seqid_validate.cpp
@@ -65,7 +65,7 @@ void CSeqIdValidate::operator()(const CSeq_id& seqId,
             foundError = true;
         }
         else
-        if (idString.size() > 50) {
+        if (idString.size() > 120) {
             description = "Local ID \"" + 
                           idString +
                           " \" exceeds 50 character limit.";
