diff --git a/c++/include/util/impl/compile_time_bits.hpp b/c++/include/util/impl/compile_time_bits.hpp
index 1c271c9..43cb126 100644
--- a/include/util/impl/compile_time_bits.hpp
+++ b/include/util/impl/compile_time_bits.hpp
@@ -35,6 +35,7 @@
  *
  */
 
+#include <cstdint>
 #include <type_traits>
 #include <limits>
 #include <stdexcept>
@@ -682,7 +683,7 @@ namespace compile_time_bits
 
         size_t get_alignment() const
         { // debug method for unit tests
-            return std::uintptr_t(m_backend.get_index()) % std::hardware_constructive_interference_size;
+            return uintptr_t(m_backend.get_index()) % std::hardware_constructive_interference_size;
         }
 
         // debug method for unit tests
