diff --git a/pyproject.toml b/pyproject.toml
index 0d32014..8d41622 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,19 +1,18 @@
 [build-system]
-requires = ["hatchling"]
-build-backend = "hatchling.build"
+requires = ["poetry-core>=1.1.0"]
+build-backend = "poetry.core.masonry.api"
 
-[project]
+[tool.poetry]
 name = "userpath"
 description = 'Cross-platform tool for adding locations to the user PATH'
 readme = "README.md"
 license = "MIT"
-requires-python = ">=3.7"
 keywords = [
   "path",
   "user path",
 ]
 authors = [
-  { name = "Ofek Lev", email = "oss@ofek.dev" },
+   "Ofek Lev <oss@ofek.dev>"
 ]
 classifiers = [
   "Development Status :: 4 - Beta",
@@ -27,22 +26,8 @@ classifiers = [
   "Programming Language :: Python :: Implementation :: CPython",
   "Programming Language :: Python :: Implementation :: PyPy",
 ]
-dependencies = [
-  "click",
-]
-dynamic = ["version"]
-
-[project.urls]
-Funding = "https://github.com/sponsors/ofek"
-History = "https://github.com/ofek/userpath/blob/master/HISTORY.rst"
-Issues = "https://github.com/ofek/userpath/issues"
-Source = "https://github.com/ofek/userpath"
-
-[project.scripts]
-userpath = "userpath.cli:userpath"
-
-[tool.hatch.version]
-path = "userpath/__init__.py"
+version = "1.8.0"
 
-[tool.hatch.build.targets.sdist]
-[tool.hatch.build.targets.wheel]
+[tool.poetry.dependencies]
+python = ">=3.7.1,<4.0.0"
+click = ">=8.0"
